src/command/vdms.rs

```diff
@@ -0,0 +1,90 @@
+//! `VDMs`: Repeat transactions on I2C3m under certain conditions.
+
+use bitfield::bitfield;
+use embedded_usb_pd::vdm::PD_VDM_MAX_SIZE;
+
+pub const CMD_VDM_LEN: usize = 31;
+pub const VDM_INITIATOR_WAIT_TIME_MS: u8 = 100;
+
+bitfield! {
+    #[derive(Clone, Copy)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub struct TaskHeader(u8);
+    impl Debug;
+
+    /// Number of VDOs to transmit
+    pub u8, num_vdo, set_num_vdo: 2, 0;
+    /// Version of the VDMs command
+    pub u8, version, set_version: 3;
+    /// SOP Target of the message in the register
+    pub u8, sop_target, set_sop_target: 5, 4;
+}
+
+impl Default for TaskHeader {
+    fn default() -> Self {
+        Self(0)
+    }
+}
+
+bitfield! {
+    #[derive(Clone, Copy)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub struct VdmsCfg(u8);
+    impl Debug;
+
+    /// initiate response
+    pub u8, init_res, set_init_res: 0;
+
+}
+
+impl Default for VdmsCfg {
+    fn default() -> Self {
+        Self(0)
+    }
+}
+
+/// The input data for the `VDMs` command.
+#[derive(Debug, Clone, Copy)]
+pub struct VdmInputData {
+    pub task_header: TaskHeader,
+    pub vdos: [u8; PD_VDM_MAX_SIZE],
```

Comment:
Since this is specific to the TPS6699x driver, and its datasheet explicitly defines the number of VDOs as 7, I'd probably hardcode it here instead of using the const from `embedded_usb_pd` (which isn't necessarily specific to this chip).
```suggestion
    pub vdos: [u8; 7],
```